api/api/models/media.py

```diff
@@ -361,6 +363,22 @@ class AbstractMediaDecisionThrough(models.Model):
     class Meta:
         abstract = True
 
+    def perform_action(self):
+        """Perform the action specified in the decision."""
+
+        if self.decision.action in {
```

Comment:
A suggestion to avoid the implicit `self.decision` query when calling `perform_action` directly (if `save` doesn't run). This will be necessary when we implement the bulk decisions anyway.

```suggestion
    def perform_action(self, action=None):
        """Perform the action specified in the decision."""
        
        action = self.decision.action if action is None else action
        
        if action in {
```

Reply:
Updated in fb8495f.